### PR TITLE
Setup puppeteer and add basic browser test

### PR DESCRIPTION
**Pull request by neelamgehlot**
### Testing

```
→ python -m tests.test-browser.run_browser_test
[C 20:52:12.650 BrowserApp]

    Copy/paste this URL into your browser when you connect for the first time,
    to login with a token:
        http://127.0.0.1:8888/lab/?token=bf0aafe1f810020b43a022a754d402e7e99615ce27557983
Starting Chrome Headless
Navigating to page: http://127.0.0.1:8888/lab/?token=bf0aafe1f810020b43a022a754d402e7e99615ce27557983
[I 20:52:13.433 BrowserApp] 302 GET /lab/?token=bf0aafe1f810020b43a022a754d402e7e99615ce27557983 (127.0.0.1) 1.71ms
Waiting for page to load...
Waiting for application to start...
Chrome test complete
[I 20:52:16.111 BrowserApp] Shutting down 0 kernels
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,4 +13,4 @@ script:
   - yarn test
   - pip install .
   - jupyter labextension install .
-  - python -m jupyterlab.browser_check
```
Comment (jaipreet-s) [inline .travis.yml]: I'd rather not remove the JupyterLab test. It's a good sanity test to ensure we are passing the JupyterLab tests and not breaking it any way (They'll keep expanding over time and we want to keep up with it, even if now we are tests are already covering that)

File: tests/test-browser/run_browser_test.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    def _browser_finished(self, future):
+        try:
+            sys.exit(future.result())
+        except Exception as e:
+            self.log.error(str(e))
+            sys.exit(1)
+
+
+def run_browser(url):
```
Comment (jaipreet-s) [inline tests/test-browser/run_browser_test.py]: If this is taken from somewhere( e.g. the JupyterLab repo) then add a link

File: tests/test-browser/chrome-test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+const assert = require('assert');
+const puppeteer = require('puppeteer');
+const inspect = require('util').inspect;
+
+const URL = process.argv[2];
+
+async function main() {
```
Comment (jaipreet-s) [inline tests/test-browser/chrome-test.js]: Would benefit from some abstraction - This way the bootstrapping code is out of the way and we're just adding more tests to the `testGit` method.

```
function main() {
   bootstrapBrowser() 
   testGit()
}

function testGit() {
  validateGitClone();
  validateGitBranch()
 ....
}
```

File: tests/test-browser/chrome-test.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  }
+}
+
+async function testApplication(page) {
+  const el = await page.waitForSelector('#browserTest', { timeout: 100000 });
+  console.log('Waiting for application to start...');
+  let testError = null;
+
+  try {
+    await page.waitForSelector('.completed');
```
Comment (saulshanabrook) [inline tests/test-browser/chrome-test.js]: What if we don't catch this here and just let the error bubble up? Then we could get rid of some code farther down.

File: tests/test-browser/chrome-test.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  } catch (e) {
+    testError = e;
+  }
+  const textContent = await el.getProperty('textContent');
+  const errors = JSON.parse(await textContent.jsonValue());
+
+  for (let error of errors) {
+    console.error(`Parsed an error from text content: ${error.message}`, error);
```
Comment (saulshanabrook) [inline tests/test-browser/chrome-test.js]: If we did this, then we could just throw an error here instead of returning it.

File: tests/test-browser/chrome-test.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+function testGitExtension(html) {
+    // Test git icons are present
+    try {
+      assert(html.includes('--jp-icon-git-clone'), 'Could not find git clone icon.');
+      assert(html.includes('--jp-icon-git-pull'), 'Could not find git pull icon.');
+      assert(html.includes('--jp-icon-git-push'), 'Could not find git push icon.');
+    } catch (e) {
+      return e;
```
Comment (saulshanabrook) [inline tests/test-browser/chrome-test.js]: Again, here what if we didn't wrap the asserts and just let the errors bubble up?

File: tests/test-browser/chrome-test.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  const html = await page.content();
+  testJupyterLabPage(html);
+
+  let testError = null;
+  testError = await testApplication(page);
+  testError = testGitExtension(html);
+
+  await browser.close();
+
+  if (testError) {
```
Comment (saulshanabrook) [inline tests/test-browser/chrome-test.js]: If we removed the error handling, we could also remove this code.

File: src/components/PastCommitNode.tsx [hunk 8]
```diff
@@ -47,22 +47,12 @@ export class PastCommitNode extends React.Component<
     };
   }
   getBranchesForCommit() {
-    const idAbrev = this.props.pastCommit.commit.slice(0, 7);
```
Comment (saulshanabrook) [inline src/components/PastCommitNode.tsx]: It seems like some older previous commits are also included in this PR, like for this pr: https://github.com/jupyterlab/jupyterlab-git/pull/280/

This isn't ideal because it makes the diff a bit longer and includes unrelated changes.

**Comment by neelamgehlot** [comment]
Sorry I accidentally closed this PR.
https://github.com/jupyterlab/jupyterlab-git/pull/297
